debug_module.tex

```diff
@@ -20,6 +20,9 @@ \chapter{Debug Module (DM)} \label{dm}
 In order to implement memory access, a target must implement either the Program
 Buffer or System Bus Access.
 
+In order to implement access to all registers, a target must implement either
```

Comment:
I thought in the past we had explicitly said it was OK to not allow access to all registers (e.g. FPRs) by any mechanism.

I don't care either way just thought we had said in the past you could make a minimalistic debug system if you wanted to.

Reply:
I'm on the fence about this.

On the one hand, accessing CSRs is essential for debugging.
On the other hand it's possible to write a small program that moves a CSR between GPRs (kind of the Program Buffer approach without a dedicated Program Buffer) so a complete debugger could be written that only explicitly supports memory and GPR access. Is anybody likely to want such a minimal implementation? It comes with a whole bunch of fragility.

Leaning towards all implementations must provide access to all registers without resorting to writing code into regular target RAM.